### PR TITLE
Expo image

### PR DESCRIPTION
**Pull request by WojtekBoman**
### Details

This PR improves the way of displaying images on mobile platforms. The most relevant changes are the removal of react-native-svg-transformer and the use expo-image to display images in the app. Previously all icons in .svg format were imported as a React components, what in result has negative influence on app's performance. Thanks to the usage of expo-image, we no longer need to transform svg icons and can use them directly by passing them to the source property of the Image component from 'expo-image' library.  

### Fixed Issues

Reduced number of rendered elements in the component tree and better rendering times.

<details>
<summary>Rendering SignInPage with react-native-svg on Android</summary>
Screenshot 2023-11-02 at 15 46 52

</details>

<details>
<summary>Rendering SignInPage with expo-image on Android</summary>
Screenshot 2023-11-02 at 15 46 15

</details>

There are not issues associated with this PR.  
$  
PROPOSAL:  
Original Problem/Solution was posted here: https://expensify.slack.com/archives/C01GTK53T8Q/p1695892593681869

### Tests

1. Go through the application and check for any problems with displaying images

- [ ] Verify that no errors appear in the JS console

### QA Steps

1. Go through the application and check for any problems with displaying images

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [ ] I linked the correct issue in the `### Fixed Issues` section above
- [ ] I wrote clear testing steps that cover the changes made in this PR
    - [ ] I added steps for local testing in the `Tests` section
    - [ ] I added steps for the expected offline behavior in the `Offline steps` section
    - [ ] I added steps for Staging and/or Production testing in the `QA steps` section
    - [ ] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [ ] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I ran the tests on **all platforms** & verified they passed on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [ ] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] I verified that if a function's arguments changed that all usages have also been updated correctly
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/software-mansion-labs/expensify-app-fork/assets/47774969/b53f75f4-add6-454d-86f8-252f8a9426f2

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/software-mansion-labs/expensify-app-fork/assets/47774969/0254fe8b-5782-47b2-86fe-8bfe534829c4

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/software-mansion-labs/expensify-app-fork/assets/47774969/32ff04ce-3cb3-4ea2-abfa-573e2949bc87

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/software-mansion-labs/expensify-app-fork/assets/47774969/c9f62989-c09c-4b5f-82ac-4297d691f31b

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/software-mansion-labs/expensify-app-fork/assets/47774969/24277678-b01e-4f3d-b4c0-f9dc60c90587

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/software-mansion-labs/expensify-app-fork/assets/47774969/731502aa-fcbe-489f-9573-8213e7beac9e
</details>

### CODE REVIEW

File: src/components/Icon/BankIcons.ts [hunk 1]
```diff
@@ -24,86 +24,154 @@ import variables from '../../styles/variables';
 type BankIcon = {
     icon: React.FC<SvgProps>;
     iconSize?: number;
+    iconName?: string;
+};
+
+type AssetIcon = {
+    icon: React.FC<SvgProps>;
```
Comment (mczernek) [inline src/components/Icon/BankIcons.ts:31]: We're using type from `react-native-svg` even though we're not really using `react-native-svg` here, am I right?

If so, let's try to find some other type here.

File: src/components/Icon/index.native.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    inline: PropTypes.bool,
+
+    // eslint-disable-next-line react/forbid-prop-types
+    additionalStyles: PropTypes.arrayOf(PropTypes.object),
+
+    /** Icon name required to create the icon test ID */
+    name: PropTypes.string,
+
+    /** Determines how the image should be resized to fit its container */
+    contentFit: PropTypes.string,
```
Comment (mczernek) [inline src/components/Icon/index.native.js:37]: Why is that necessary on native and not on web?

File: src/components/Icon/index.native.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        const height = this.props.small ? variables.iconSizeSmall : this.props.height;
+        const iconStyles = [StyleUtils.getWidthAndHeightStyle(width, height), IconWrapperStyles, styles.pAbsolute, ...this.props.additionalStyles];
+        if (this.props.inline) {
+            return (
+                <View
+                    testID={`Svg${this.props.name} Icon`}
+                    style={[StyleUtils.getWidthAndHeightStyle(width, height), styles.bgTransparent, styles.overflowVisible]}
+                >
+                    <View style={iconStyles}>
+                        <Image
```
Comment (mczernek) [inline src/components/Icon/index.native.js:69]: It seems that we could extract this component (and `IconComponent` in index.js) to two seperate files (one with `native.js` extension) and keep divergence between platforms at its minimum.

File: src/components/Icon/index.native.js (hunk 2 above)
Comment (WojtekBoman) [reply src/components/Icon/index.native.js:37]: On web we still use @svgr/webpack to transform svg icons to components, icons displayed as a components don't need to be additionally styled

File: src/components/AvatarCropModal/AvatarCropModal.js [hunk 4]
```diff
@@ -44,7 +45,7 @@ const propTypes = {
     isVisible: PropTypes.bool.isRequired,
 
     /** Image crop vector mask */
-    maskImage: PropTypes.func,
+    maskImage: PropTypes.oneOfType([PropTypes.func, imagePropTypes.source]),
```
Comment (mczernek) [inline src/components/AvatarCropModal/AvatarCropModal.js:48]: Seems that this exact type is used in several places to indicate a field that accepts image component - maybe we could define this type within `imagePropTypes` and reuse it?

File: src/components/CardPreview/index.native.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    },
+};
+
+function CardPreview({privatePersonalDetails: {legalFirstName, legalLastName}, session: {email}}) {
+    usePrivatePersonalDetails();
+    const cardHolder = legalFirstName && legalLastName ? `${legalFirstName} ${legalLastName}` : email;
+
+    return (
+        <View style={styles.walletCard}>
+            <Image
```
Comment (mczernek) [inline src/components/CardPreview/index.native.js:42]: Same as in other places, even though only this particular component differs between native and web, we have two big files that are mostly identical :(

Maybe we could create some component called `<Image>` of our own and have two implementations of it depending on platform?

File: src/components/Image/index.native.js [hunk 6]
```diff
@@ -34,17 +34,17 @@ function Image(props) {
...
             {...rest}
             source={imageSource}
-            onLoad={(evt) => {
-                const {width, height} = evt.nativeEvent;
-                dimensionsCache.set(source.uri, {width, height});
-                if (props.onLoad) {
-                    props.onLoad(evt);
-                }
-            }}
+            // onLoad={(evt) => {
```
Comment (mczernek) [inline src/components/Image/index.native.js:41]: Remove usuned code

File: src/components/QRShare/index.native.js [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    }
+
+    render() {
+        return (
+            <View
+                style={styles.shareCodeContainer}
+                onLayout={this.onLayout}
+            >
+                <View style={styles.expensifyQrLogo}>
+                    <Image
```
Comment (mczernek) [inline src/components/QRShare/index.native.js:53]: ANother place with two almost identical files.

File: src/components/ValidateCode/JustSignedInModal.js [hunk 8]
```diff
@@ -23,9 +23,11 @@ function JustSignedInModal(props) {
             <View style={styles.deeplinkWrapperMessage}>
                 <View style={styles.mb2}>
                     <Icon
+                        displayInDefaultIconColor
                         width={variables.modalTopIconWidth}
                         height={props.is2FARequired ? variables.modalTopIconHeight : variables.modalTopBigIconHeight}
                         src={props.is2FARequired ? Illustrations.SafeBlue : Illustrations.Abracadabra}
+                        name={props.is2FARequired ? 'Safe' : 'Abracadabra'}
```
Comment (mczernek) [inline src/components/ValidateCode/JustSignedInModal.js:30]: Is Abracadabra intentional? :D

File: src/components/MentionSuggestions.js [hunk 9]
```diff
@@ -77,7 +76,7 @@ function MentionSuggestions(props) {
                         size={isIcon ? CONST.AVATAR_SIZE.MENTION_ICON : CONST.AVATAR_SIZE.SMALLER}
                         name={item.icons[0].name}
                         type={item.icons[0].type}
-                        fill={themeColors.success}
+                        // fill={themeColors.success}
```
Comment (mczernek) [inline src/components/MentionSuggestions.js:79]: Unused code

File: src/components/MultipleAvatars.js [hunk 10]
```diff
@@ -137,7 +138,7 @@ function MultipleAvatars(props) {
                     <Avatar
                         source={props.icons[0].source}
                         size={props.size}
-                        fill={themeColors.iconSuccessFill}
+                        // fill={themeColors.iconSuccessFill}
```
Comment (mczernek) [inline src/components/MultipleAvatars.js:141]: Unused code

File: src/components/RoomHeaderAvatars.js [hunk 11]
```diff
@@ -45,7 +44,7 @@ function RoomHeaderAvatars(props) {
                         <Avatar
                             source={props.icons[0].source}
                             imageStyles={[styles.avatarLarge]}
-                            fill={themeColors.iconSuccessFill}
+                            // fill={themeColors.iconSuccessFill}
```
Comment (mczernek) [inline src/components/RoomHeaderAvatars.js:47]: Unused code - 2 places in this file

File: src/components/SubscriptAvatar.js [hunk 12]
```diff
@@ -80,8 +80,9 @@ function SubscriptAvatar({size, backgroundColor, mainAvatar, secondaryAvatar, no
                             StyleUtils.getBorderColorStyle(backgroundColor),
                         ]}
                         source={secondaryAvatar.source}
+                        avatarImageName={secondaryAvatar.avatarImageName}
                         size={isSmall ? CONST.AVATAR_SIZE.SMALL_SUBSCRIPT : CONST.AVATAR_SIZE.SUBSCRIPT}
-                        fill={themeColors.iconSuccessFill}
+                        // fill={themeColors.iconSuccessFill}
```
Comment (mczernek) [inline src/components/SubscriptAvatar.js:85]: Unused code

File: src/pages/ErrorPage/GenericErrorPage/index.js [hunk 13]
```diff
@@ -33,6 +33,7 @@ function GenericErrorPage({translate}) {
                             <View style={styles.mb5}>
                                 <Icon
                                     src={Expensicons.Bug}
+                                    name="Bug"
```
Comment (mczernek) [inline src/pages/ErrorPage/GenericErrorPage/index.js:36]: Very similar story again

File: src/pages/iou/ReceiptDropUI/index.native.js [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+    receiptImageTopPosition: 0,
+};
+
+function ReceiptDropUI({onDrop, receiptImageTopPosition}) {
+    const {translate} = useLocalize();
+    return (
+        <DragAndDropConsumer onDrop={onDrop}>
+            <View style={[styles.receiptDropOverlay, styles.w100, styles.h100, styles.justifyContentCenter, styles.alignItemsCenter]}>
+                <View style={styles.receiptImageWrapper(receiptImageTopPosition)}>
+                    <Image
```
Comment (mczernek) [inline src/pages/iou/ReceiptDropUI/index.native.js:29]: SAme story - almost identical files

File: src/pages/settings/AboutPage/index.native.js [hunk 15]
```diff
@@ -0,0 +1,170 @@
...
+                <>
+                    <HeaderWithBackButton
+                        title={props.translate('initialSettingsPage.about')}
+                        onBackButtonPress={() => Navigation.goBack(ROUTES.SETTINGS)}
+                    />
+                    <ScrollView contentContainerStyle={[styles.flexGrow1, styles.flexColumn, styles.justifyContentBetween, safeAreaPaddingBottomStyle]}>
+                        <View style={[styles.flex1]}>
+                            <View style={styles.pageWrapper}>
+                                <View style={[styles.settingsPageBody, styles.mb6, styles.alignItemsCenter]}>
+                                    <Image
```
Comment (mczernek) [inline src/pages/settings/AboutPage/index.native.js:107]: Once again, this is the only difference between this and `index.js`

File: tests/unit/SidebarOrderTest.js [hunk 16]
```diff
@@ -182,7 +182,7 @@ describe('Sidebar', () => {
                     // Then there should be a pencil icon and report one should be the first one because putting a draft on the active report should change its location
                     // in the ordered list
                     .then(() => {
-                        const pencilIcon = screen.queryAllByTestId('Pencil Icon');
+                        const pencilIcon = screen.queryAllByTestId('SvgPencil Icon');
```
Comment (mczernek) [inline tests/unit/SidebarOrderTest.js:185]: Are those the only two test files that uses images test ids?

File: src/components/Icon/index.native.js [hunk 17]
```diff
@@ -0,0 +1,99 @@
...
+    additionalStyles: PropTypes.arrayOf(PropTypes.object),
+
+    /** Icon name required to create the icon test ID */
+    name: PropTypes.string,
+
+    /** Determines how the image should be resized to fit its container */
+    contentFit: PropTypes.string,
+
+    /** Is icon displayed in its own color */
+    displayInDefaultIconColor: PropTypes.bool,
```
[the remark this answers is not in the document]
Comment (WojtekBoman) [reply src/components/Icon/index.native.js:40]: But If I want to override it I have to pass null value through the fill property. Is it ok? I've added this property to make code more readable. When we want to display icon in original color, component should not have defined tintColor property or this property's value must be null. Instead of passing null through the fill property I created this prop.

[the remark this answers is not in the document]
Comment (mczernek) [reply src/components/Icon/index.native.js:40]: Is there any place where we want to display Icon's color defined in SVG file without any tinting (either with color from theme or passed as property)? If so, why does it work on web without this additional boolean? If no, I think that passing `undefined` as `fill` is fine

[the remark this answers is not in the document]
Comment (WojtekBoman) [reply src/components/Icon/index.native.js:40]: Fill prop behaviour is different when we use expo-image because in this case we set tintColor for whole image, null value allows to use original icon color. When we pass this prop to component transformed by svg-transfomer we can set fill prop's value directly in svg asset, it doesn't set this color for whole icon but only for the part which should be displayed in this color. It can be noticed on logo asset which i modified in this PR. I set fill prop's value directly in svg asset and it changed only the logo part of this asset not whole image, small icon which shows environment didn't change

**Comment by blazejkustra** [comment]
@BartoszGrajdek To mialo zostac zmergowane na naszym forku? 😅
